doc/developer/index.adoc

```diff
@@ -26,29 +26,136 @@ These conditions are to ensure that existing user code will continue to work whe
+For this reason approved Pull Requests may stay open for some time until a
+release plan that incorporates these changes is in place.
 
 == Patch releases
 
-Patch releases are created by cherry-picking desired bug fixes from the master branch into a release branch, and when complete, tagging this release branch with the version number before release deployment. Once release deployment is complete, the release branch is merged back into the master branch, and then deleted.
+Patch releases are created by branching the _master_ branch into a release branch, and
+when complete, tagging this release branch with the version number before
+release deployment. Once release deployment is complete, the release branch
+is deleted.
+
+IMPORTANT: It is important that the _master_ branch is always in a release ready state, so
+a patch release from the _master_ branch can be done in an ad-hoc fashion,
+without the need for a formal review.
+
+Plans to do a patch release are announced by the project lead on the
+https://dev.eclipse.org/mailman/listinfo/rdf4j-dev[rdf4j-dev@eclipse.org mailinglist],
+usually about a week in advance, with an open invitation for contributors to
+propose additional fixes to include, which are done as Pull Requests to the
+_master_ branch.
 
-A patch release can be done in an ad-hoc fashion, without the need for a formal review. Plans to do a patch release are announced by the project lead on the https://dev.eclipse.org/mailman/listinfo/rdf4j-dev[rdf4j-dev@eclipse.org mailinglist], usually about a week in advance, with an open invitation for contributers to propose additional fixes to include.
+=== Creating a patch release branch
 
-=== Creating a release branch
+Any fixes to be included in the a patch release must be merged into the _master_
+branch first.
+A patch release branch should differ from the _master_ branch, at the time of
+release, only by the version number - a patch release branch has a patch number
+version, while the _master_ branch has a SNAPSHOT version.
+To create a patch release branch, follow these steps:
+
+1. Check out the _master_ branch.
++
+E.g. if we're preparing a release 2.2.1, the _master_ branch will have the version 2.2-SNAPSHOT:
++
+    git checkout master
 
-A patch release uses its immediate predecessor release as its basis. This means we need to create a release branch not by simply branching from the current master branch, but by branching from a specific release tag. To create a patch release branch, follow these steps:
+2. Create a new release branch, named `releases/<version>`:
++
+    git checkout -b releases/2.2.1
+
+3. Fix maven version numbers in the release branch.
++
+We need to set the project version to `2.2.1` by using:
++
+    mvn versions:set
++
+This will ask for the new version number. Enter `2.2.1` to indicate that this is the release branch for 2.2.1 release.
```

Comment:
Even if the release branch is done locally, unless you squash it down to a single commit before tagging and pushing, it would be useful to keep the -SNAPSHOT version until the actual commit that will be tagged. Squashing before tagging and pushing is definitely an option if the release manager is doing this all on their machine but would require some more instructions here.